packages/nextjs/src/tools/codegen/import-map.ts

```diff
@@ -0,0 +1,413 @@
+import * as ts from 'typescript';
+import path from 'path';
+import fs from 'fs';
+import { debug } from '@sitecore-content-sdk/core';
+import { getComponentList } from '@sitecore-content-sdk/core/tools';
+import { SitecoreConfig } from '@sitecore-content-sdk/core/config';
+import crypto from 'crypto';
+import { defaultImportEntries } from './default-import-map';
+import { ImportEntry } from '@sitecore-content-sdk/core/editing';
+
+let _defaultImportEntries = defaultImportEntries;
+let _getComponentList = getComponentList;
+
+export const unitMocks = ({
+  mockDefaultImportEntries,
+  getComponentListStub,
+}: {
+  mockDefaultImportEntries?: ImportEntry[];
+  getComponentListStub?: typeof getComponentList;
+}) => {
+  mockDefaultImportEntries && (_defaultImportEntries = mockDefaultImportEntries);
+  getComponentListStub && (_getComponentList = getComponentListStub);
+};
+
+type ExportDefinition = {
+  name: string;
+  value: string;
+};
+
+/**
+ * Type describing import map entry in final file
+ * Represents structure i.e.
+ * {module: '/import-path',
+ * exports: [{ name: 'myName1', value: myName1 },..]}
+ * Note that string name and implementation (value) name would be the same
+ */
+export type ImportMapEntry = {
+  module: string;
+  namedExports: ExportDefinition[];
+  defaultExports: ExportDefinition[];
+};
+
+/**
+ * Args for import map generation
+ * Specifies paths to include and exclude when generating imports
+ */
+export type WriteImportMapArgs = {
+  paths: string[];
+  exclude?: string[];
+};
+
+/**
+ * Import names definition
+ * @typedef ImportName
+ * @property {string} name - the original name of the import
+ * @property {boolean} [isWildcard] - if import is a wildcard import (import * as name from 'module')
+ */
+export type ImportName = {
+  name: string;
+  isWildcard?: boolean;
+};
+
+/**
+ * Import module definition, specifying all the imports for a given import path
+ */
+export type ModuleExports = {
+  // import { import1, import2 as imp3 }
+  namedExports: Map<string, string>;
+  // import coolName or import * as coolName or import coolName as coolerName
+  defaultExports: Map<string, string>;
+};
+
+/**
+ * Gets an import string and outputs import info with aliases, if present
+ * @param {ts.ImportDeclaration} importNode import definition node
+ * @returns {ImportName[] | ImportName | null} object(s) with definition name
+ */
+export const getImportedValues = (
+  importNode: ts.ImportDeclaration
+): ImportName[] | ImportName | null => {
+  const importClause = importNode.importClause?.getText();
+  const aliasImport = /^([a-zA-Z0-9]+) as .+$/g;
+  const namespaceImport = /^\* as (.+)$/g;
+
+  if (!importClause) {
+    console.warn('Cannot parse import string from: %s', importNode.getText());
+    return null;
+  }
+  if (/^\{.*\}$/.test(importClause)) {
```

Comment:
It seems this code still doesn't work well with compound import declaration.
default import + namespace import
default import + named imports